platform/darwin/src/MGLStyleAttributeValue.h

```diff
@@ -0,0 +1,31 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLTypes.h"
+
+typedef struct {
+    float red;
+    float green;
+    float blue;
+    float alpha;
+} MGLAttributeColor;
```

Comment:
Instead of defining our own color struct type, why not extend NSColor and UIColor to conform to MGLStyleAttributeValue? Then `-[MGLStyleAttributeValue colorValue]` would return an instance of NSColor or UIColor (depending on the platform, using an ifdef).